servers/src/main/java/tachyon/worker/block/allocator/GreedyAllocator.java

```diff
@@ -27,7 +28,7 @@
  * A greedy allocator that returns the first Storage dir fitting the size of block to allocate.
  * This class serves as an example how to implement an allocator.
  */
-public class GreedyAllocator implements Allocator {
+public class GreedyAllocator extends BlockStoreEventListenerBase implements Allocator {
```

Comment:
@calvinjia +1 for keeping it an interface, I will close this PR and the JIRA, thanks!